### ISSUE TITLE
add under5 variable for HC CTC expansion

### ISSUE THREAD

**Issue by MattHJensen** [open-source-economics/taxdata#42]
@codykallen recently implemented a quick version of the Clinton child tax credit expansion for Alex Brill's [blog post](https://www.aei.org/publication/6-things-to-know-about-hillary-clintons-new-child-tax-credit-expansion/) and [op ed ](http://www.realclearmarkets.com/articles/2016/10/28/who_will_pay_for_hillary_clintons_tax-code_social_engineering_102402.html) on the policy. 

I recently asked him how he did it, and here's what he said:
(@codykallen, thanks for permission to reproduce this here)

> 
> The code is available at https://github.com/codykallen/Tax-Calculator/blob/ctc_expansion/taxcalc/functions.py
>  
> I implemented this using the personal credit, and randomly assigning whether a child is under 5. The necessary changes to functions.py are as follows:
```
In the AGI function:
-          personal_credit = II_credit[MARS - 1]
-          if II_credit_prt > 0. and c00100 > II_credit_ps[MARS - 1]:
-          credit_phaseout = II_credit_prt * (c00100 - II_credit_ps[MARS - 1])
-          personal_credit = max(0., personal_credit - credit_phaseout)
+     under5 = 0
+  if n24 > 0:
+         for i in range(n24):
+             if random.random() < 0.273:
+                    under5 += 1
+  personal_credit = II_credit[MARS - 1] * under5
Don’t forget to include n24 in the inputs.
 
In the ChildTaxCredit function:
-          prectc = CTC_c * n24
+     prectc = CTC_c * n24 + personal_credit
Don’t forget to include personal_credit in the inputs.
 
In the AdditionalCTC function:
-          c82890 = ACTC_rt * c82885
+     if personal_credit > 0:
+      c82890 = 0.45 * c82885
+  else:
+      c82890 = ACTC_rt * c82885
Don’t forget to include personal_credit in the inputs.
 
In the IITAX function:
-          _refund = c59660 + c11070 + c10960 + personal_credit
+     _refund = c59660 + c11070 + c10960
```
> You do not need to keep personal_credit in the inputs, but that won’t affect the ability to function (note that it will cause a test to fail).
>  
>  
> Since I didn’t have data on which children are under 5, I just applied I random probability for each child being under 5 based on government estimates of child age distributions. This should let you simulate the effect of the CTC expansion for families with children under 5. Under the proposal, children under 5 get a $2000 CTC instead of $1000, the ACTC refundability threshold is lowered to zero, and families with children under 5 get a 45% refundability rate instead of 15%.

@andersonfrailey, could you add an "under5" variable to the puf? 

I hope to implement the CTC expansion in TC w/o hijacking the personal credit.

**Comment by andersonfrailey** [comment]
@MattHJensen the PUF is ready. 

Rather than randomly assign a variable like @codykallen did I was able to take the `cps-puf.csv` file before the `agedp*` variables were dropped and added up the number of children under 5 for each tax unit. 

Do you want me to fully redistribute this PUF now or wait and see if there's anything else that needs to be added and just send it to you for now?

**Comment by MattHJensen** [comment]
Thanks! Why don't you distribute the file to everyone as a preliminary version. 

Will you have a PR that demonstrates these changes?

**Comment by andersonfrailey** [comment]
@MattHJensen opening one up now.

**Comment by martinholmer** [comment]
Closing taxdata issue #42 because it has been resolved by merged taxdata pull request #43 that adds the `nu05` variable to the `puf.csv` file.

**Comment by martinholmer** [comment]
@MattHJensen,  Do you want to close taxdata issue #42?  It seems to be resolved.

**Comment by martinholmer** [comment]
taxdata pull request #43 resolves issue #42, right?   Unless somebody corrects my understanding, I'll close this issue on Monday, Feb 6th.

**Comment by andersonfrailey** [comment]
@martinholmer correct. This issue can be closed.

**Comment by martinholmer** [comment]
@andersonfrailey, Thanks for the quick confirmation.  Closing resolved issue #42.

**Comment by martinholmer** [comment]
Oops!  Guess I don't have permission to close an issue in this repository.

@MattHJensen
